spectrwm.c

```diff
@@ -5567,7 +5655,7 @@ focus_region(struct swm_region *r)
 		set_region(r);
 
 		/* New region is empty; need to manually unfocus win. */
-		if (old_r) {
+		if (old_r != r) {
```

Comment:
@LordReg, this change causes spectrwm to segfault upon starting on OpenBSD amd64 current:

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  focus_region (r=0x87ab32ba9c0) at spectrwm.c:5659
5659                            unfocus_win(old_r->ws->focus);
(gdb) bt full
#0  focus_region (r=0x87ab32ba9c0) at spectrwm.c:5659
        none = <error reading variable none (Cannot access memory at address 0x0)>
        nfw = <optimized out>
        old_r = 0x0
#1  0x00000878a0ad6d97 in setup_focus () at spectrwm.c:14610
        r = 0x87ab32ba9c0
        w = <optimized out>
#2  0x00000878a0ad87e8 in main (argc=<optimized out>, argv=<optimized out>) at spectrwm.c:15343
        pfd = {{fd = 0, events = 0, revents = 0}, {fd = 0, events = 0, revents = 0}}
        sact = {__sigaction_u = {__sa_handler = 0x878a0ab3ea0 <sighdlr>,
            __sa_sigaction = 0x878a0ab3ea0 <sighdlr>}, sa_mask = 0, sa_flags = 8}
        stdin_ready = <error reading variable stdin_ready (Cannot access memory at address 0x0)>
        cfile = <optimized out>
        ch = <optimized out>
        evt = <optimized out>
        mne = <optimized out>
        i = <optimized out>
        num_screens = 1
        r = <optimized out>
        num_readable = <optimized out>
```

Reverting this bit fixes the above for me.

Reply:
Thanks for the report, fixed.